packages/nimble-components/src/wafer-map/modules/computations.module.ts

```diff
@@ -0,0 +1,189 @@
+import { scaleBand, ScaleLinear, scaleLinear } from 'd3-scale';
+import type { WaferMapDie } from '../types';
+import { Dimensions, Margin, WaferMapQuadrant } from '../types';
+
+interface MapDimensions {
+    origin: {
+        x: number,
+        y: number
+    };
+    rows: number;
+    cols: number;
+}
+
+/**
+ * Computations module
+ */
+export class Computations {
+    public containerDimensions: Dimensions;
+    public dieDimensions: Dimensions = { width: 0, height: 0 };
+    public radius: number;
+    public margin: Margin;
+
+    public horizontalScale: ScaleLinear<number, number>;
+    public verticalScale: ScaleLinear<number, number>;
+
+    private readonly baseMargin: Margin = {
+        top: 20,
+        right: 20,
+        bottom: 20,
+        left: 20
+    };
```

Comment:
Looks like this is a default value kinda reference / lookup table. We can mark it `as const` to prevent accidental changes

```suggestion
    private readonly baseMargin: Margin = {
        top: 20,
        right: 20,
        bottom: 20,
        left: 20
    } as const;
```